tcl/file.tcl

```diff
@@ -86,10 +86,13 @@ proc qc::cat {filename} {
 
 proc qc::file2url {file} {
     # Takes a file and returns url path relative to www root.
-    if { [regexp "^[ns_pagepath](.+)\$" $file -> url] } {
-        return $url
+    set root [ns_pagepath]
+    if { [string equal -length [string length $root] $root $file]
```

Comment:
Not essential to this bug fix, but `ns_pagepath` isn't guaranteed to be regex-safe, and we don't have a generic regex-escape proc at the moment.